index.bs

```diff
@@ -900,77 +900,104 @@ See [[#naming-is-hard]] for general (cross-language) advice on naming.
-changes in this timing. But user agents differ significantly in both the timing of garbage
-collection and whether certain objects are collected at all, which means the resulting code will be
-non-interoperable. Worse, according to the usual rules of game theory as applied to browsers, this
-kind of scenario could force other user agents to copy the garbage collection timing of the
-original in order to create interoperability. This would cause current garbage collection
-strategies to ossify, preventing improvement in one of the most dynamic areas of JavaScript
-virtual machine technology.
-
-In particular, this means that you shouldn't expose any API that acts as a weak reference, e.g. with a
-property that becomes <code highlight="js">null</code> once garbage collection runs. Such freeing of
-memory should be entirely deterministic.
-
-For example, currently, {{Document/getElementsByTagName}} permits reusing the {{HTMLCollection}}
-which it creates, when it's called with the same receiver and tag name. In practice, engines reuse
-the output if it has not been garbage collected. This creates behavior which differs based on the
-details of garbage collection, which is strongly discouraged. If {{Document/getElementsByTagName}}
-were designed today, the advice to the specification authors would be to either reliably reuse
-the output, or to produce a new {{HTMLCollection}} each time it's invoked.
-
-The case of {{Document/getElementsByTagName}} is particularly insidious because nothing about
-the API visibly relates to garbage collection details. By contrast, special APIs for this particular
-purpose such as `WeakRef` and `FinalizationGroup` have their GC interaction as a specific part of
-their contract with the developer.
+Don't modify data accessed from JavaScript APIs
+while a JavaScript <a>event loop</a> is running.
+
+A JavaScript Web API is generally a wrapper around
+a feature implemented in a lower-level language,
+such as C++ or Rust.
+Unlike those languages,
+Developers can expect when using JavaScript 
+that once a piece of code begins executing,
+it will continue executing until it has completed.
+
+Because of that, JavaScript authors take for granted
+that the data available to a function won’t change unexpectedly
+while the function is running.
+
+So if a JavaScript Web API exposes some piece of data,
+such as an object property,
+the user agent must not update that data
+while a JavaScript task is running.
+Instead, if the underlying data changes,
+<a>queue a task</a> to modify the exposed version of the data.
+
+<div class="example">
+If a JavaScript task has accessed the {{navigator.onLine}} property,
+and browser's online status changes,
+the property won't be updated until the next task runs.
+</div>
+
+<h3 id="js-gc">Don't expose garbage collection</h3>
+
+Ensure your JavaScript Web APIs don't behave differently
+depending on the timing of garbage collection.
```

Comment:
I think this statement is semantically different from the opening sentence of the original text.